packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -793,9 +793,11 @@ class DropdownButton<T> extends StatefulWidget {
   /// The [items] must have distinct values. If [value] isn't null then it
   /// must be equal to one of the [DropdownMenuItem] values. If [items] or
   /// [onChanged] is null, the button will be disabled, the down arrow
-  /// will be greyed out, and the [disabledHint] will be shown (if provided).
-  /// If [disabledHint] is null and [hint] is non-null, [hint] will instead be
-  /// shown.
+  /// will be greyed out.
+  ///
+  /// If [value] is null, [hint] or [disabledHint] is non-null, depends on the
+  /// dropdown button enabled or disabled status, the [hint] or [disabledHint] widget
+  /// is displayed as a placeholder for the dropdown button's value.
```

Comment:
We can split this up into enabled/disabled statuses to make it easier to understand:
```suggestion
  /// If [value] is null and the button is enabled, [hint] will be displayed if it is non-null.
  ///
  /// If [value] is null and the button is disabled, [disabledHint] will be displayed if it is non-null.
  /// If [disabledHint] is null, then [hint] will be displayed if it it non-null.
```